libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -509,17 +510,11 @@ - (void)continueNativeBrowserFlow {
     // Launch the native browser.
     [self log:SFLogLevelDebug format:@"%@: Initiating native browser flow with URL %@", NSStringFromSelector(_cmd), approvalUrl];
     NSURL *nativeBrowserUrl = [NSURL URLWithString:approvalUrl];
-    BOOL browserOpenSucceeded = [SFApplicationHelper openURL:nativeBrowserUrl];
-    if (!browserOpenSucceeded) {
-        [self log:SFLogLevelError format:@"%@: Could not launch native browser with URL %@", NSStringFromSelector(_cmd), approvalUrl];
-        NSError *launchError = [[self class] errorWithType:kSFOAuthErrorTypeBrowserLaunchFailed description:@"The native browser failed to launch for advanced authentication."];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [self notifyDelegateOfFailure:launchError authInfo:self.authInfo];
-        });
-    } else {
-        self.advancedAuthState = SFOAuthAdvancedAuthStateBrowserRequestInitiated;
-    }
-    
+
+    SFSafariViewController *svc = [[SFSafariViewController alloc] initWithURL:nativeBrowserUrl];
+    svc.delegate = self;
+    [[SFRootViewManager sharedManager] pushViewController:svc];
```

Comment:
This breaks the configurability of root view management in the SDK.  Everywhere else where `SFRootViewManager` is employed, there is configurable logic to replace it with an alternative approach to "uber" VC management.

We need to provide a mechanism for configuring how the VC will be displayed.  I would suggest:

1. Creating an `SFOAuthCoordinatorDelegate` method (similar to `oauthCoordinator:didBeginAuthenticationWithView:`) that hands this VC off to `SFAuthenticationManager`.
2. Providing customization hooks similar to (or embedded in) `SFAuthenticationViewHandler` within `SFAuthenticationManager`, to allow customizable presentation of the VC.